api/registrations/serializers.py

```diff
@@ -314,8 +314,13 @@ class RegistrationFileSerializer(FileSerializer):
     comments = FileCommentRelationshipField(related_view='registrations:registration-comments',
                                             related_view_kwargs={'node_id': '<node._id>'},
                                             related_meta={'unread': 'get_unread_comments_count'},
-                                            filter={'target': 'get_file_guid'})
+                                            filter={'target': 'get_file_guid'}
+                                            )
 
+    node = RelationshipField(related_view='registrations:registration-detail',
```

Comment:
Hmmm. I wonder what happens to sub-views on a withdrawn registration. https://api.osf.io/v2/registrations/28yxd/files/ apparently permissions work properly, so you shouldn't have access to a file that belongs to a withdrawn registration. Phew. Very good, continue on.